protostar/starknet_gateway/gateway_facade.py

```diff
@@ -218,11 +218,21 @@ async def declare(
         account_client = await self._create_account_client(
             account_address=account_address, signer=signer
         )
-        declare_tx = await account_client.sign_declare_transaction(
-            compiled_contract=compiled_contract,
-            max_fee=max_fee if isinstance(max_fee, int) else None,
-            auto_estimate=max_fee == "auto",
-        )
+
+        try:
+            declare_tx = await account_client.sign_declare_transaction(
+                compiled_contract=compiled_contract,
+                max_fee=max_fee if isinstance(max_fee, int) else None,
+                auto_estimate=max_fee == "auto",
+            )
+        except ClientError as ex:
+            message = (
+                "No account associated with provided account address found. Contact your wallet provider."
+                if "StarknetErrorCode.UNINITIALIZED_CONTRACT" in ex.message
```

Comment:
I think we want to also check for address of the account, matching the one given in the error.

Reply:
Fine, but it can only be done by searching through the ex.message string :/

Reply:
I think this is good enough